dependencies.go

```diff
@@ -12,56 +13,74 @@ import (
 	"github.com/influxdata/flux/internal/errors"
 )
 
-const InterpreterDepsKey = "interpreter"
+// Dependency is an interface that must be implemented by every injectable dependency.
+// On Inject, the dependency is injected into the context and the resulting one is returned.
+// Every dependency must provide a function to extract it from the context.
+type Dependency interface {
+	Inject(ctx context.Context) context.Context
+}
+
+type key int
+
+const dependenciesKey key = iota
 
-type Interface interface {
+type Dependencies interface {
```

Comment:
I'd like to refactor this later, but I think this is the correct way to do this for now.